File: Professional Image Modifier 5/[VB]-Professional Image Modifier 5/VB/ImageFunctions/Controls/FASTCornerProperties.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace ImageFunctions.Controls
	Partial Public Class FASTCornerProperties
		Inherits UserControl

		Public Property SupressDefault() As Boolean
		Public Property ThresholdDefault() As Integer
		Public Property Supress() As Boolean
		Public Property Threshold() As Integer

		''' <summary>
		''' The Defaults are not real defaults - you should investigate appropriate defaults for Threshold and Sigma
		''' </summary>
		Public Sub New()
			InitializeComponent()
			ThresholdDefault = 40
			SupressDefault = True
			trackThreshold.Value = ThresholdDefault
			cbSupress.Checked = SupressDefault
		End Sub

		Public Sub SetDefaults()
			trackThreshold.Value = ThresholdDefault
			cbSupress.Checked = SupressDefault
		End Sub


		Private Sub trackThreshold_ValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles trackThreshold.ValueChanged
			lblThreshold.Text = trackThreshold.Value.ToString()
			Threshold = trackThreshold.Value
		End Sub

		Private Sub cbSupress_CheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbSupress.CheckStateChanged
			Supress = cbSupress.Checked
		End Sub
	End Class
End Namespace

File: Professional Image Modifier 4/[C#]-Professional Image Modifier 4/C#/ImageFunctions/Controls/FASTCornerProperties.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ImageFunctions.Controls
{
	public partial class FASTCornerProperties : UserControl
	{
		public bool SupressDefault { get; set; } // As set by Aforge
		public int ThresholdDefault { get; set; }  // As set by Aforge
		public bool Supress { get; set; } // Set by User
		public int Threshold { get; set; } // Set by User

		/// <summary>
		/// The Defaults are not real defaults - you should investigate appropriate defaults for Threshold and Sigma
		/// </summary>
		public FASTCornerProperties()
		{
			InitializeComponent();
			ThresholdDefault = 40;
			SupressDefault = true;
			trackThreshold.Value = ThresholdDefault;
			cbSupress.Checked = SupressDefault;
		}

		public void SetDefaults()
		{
			trackThreshold.Value = ThresholdDefault;
			cbSupress.Checked = SupressDefault;
		}


		private void trackThreshold_ValueChanged(object sender, EventArgs e)
		{
			lblThreshold.Text = trackThreshold.Value.ToString();
			Threshold = trackThreshold.Value;
		}

		private void cbSupress_CheckStateChanged(object sender, EventArgs e)
		{
			Supress = cbSupress.Checked;
		}
	}
}
